shared/js/background/message-handlers.js

```diff
@@ -65,32 +65,60 @@ export function getBrowser () {
 }
 
 /**
- * Send a broken site report to the pixel endpoint.
- * @param {string} brokenSiteParams Search param string to send to the broken site pixel endpoint
- * @returns {void}
+ * Only the dashboard sends this message, so we import the types from there.
+ * @param {import('@duckduckgo/privacy-dashboard/schema/__generated__/schema.types').BreakageReportRequest} breakageReport
+ * @returns {Promise<void>}
  */
-export function submitBrokenSiteReport (brokenSiteParams) {
-    return brokenSiteReport.fire(brokenSiteParams)
+export async function submitBrokenSiteReport (breakageReport) {
+    const { category, description } = breakageReport
+
+    const currentTab = await utils.getCurrentTab()
+    if (!currentTab?.id) {
+        console.error('could not access the current tab...')
+        return
+    }
+
+    const tab = await getTab(currentTab.id)
+    if (!tab) {
+        console.error('cannot access current tab with ID ' + currentTab.id)
+        return
+    }
+    const tsd = settings.getSetting('tds-etag')
+    return breakageReportForTab(tab, tsd, category, description)
 }
 
-export async function getTab (tabId) {
+/**
+ * @param tabId
+ * @returns {Promise<import("./classes/tab.es6")>}
+ */
+async function getTab (tabId) {
     // Await for storage to be ready; this happens on service worker closing mostly.
     await settings.ready()
     await tdsStorage.ready('config')
-
-    const tab = await tabManager.getOrRestoreTab(tabId)
-    return new LegacyTabTransfer(tab)
+    return tabManager.getOrRestoreTab(tabId)
```

Comment:
`getOrRestoreTab` perhaps could have these two things to wait on instead; I think the `getTab` message handler isn't used now right?